client/normal_value.go

```diff
@@ -0,0 +1,1531 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package client
+
+import (
+	"time"
+
+	"github.com/sourcenetwork/immutable"
+	"golang.org/x/exp/constraints"
+)
+
+type NormalValue interface {
```

Comment:
todo: Please document all the public stuff before merge.  If you end up breaking up this file, a doc.go with package level documentation might be nice too.

Reply:
will do